docs/contents/releases.md

```diff
@@ -1,128 +1,159 @@
+    generator actually imports the ACK common runtime, this dependency exists
+    because the Go code produced by the templates inside the code generator imports
+    the ACK common runtime. In order to make this Go code dependency more strict,
+    we have a test package inside the ACK code generator that imports the ACK
+    common runtime. In this way, we're able to include a version-specific
+    dependency line in the ACK code generator's `go.mod` file, thereby allowing
+    Go's module infrastructure to pin the dependency between the code generator and
+    the common runtime.
 
-The `PLANNED` status indicates that **the AWS service is on our radar** for
-building into ACK.
+## Stable Helm Charts
 
-There will be a Github Issue for tracking the build of the ACK service
-controller for the service.
+We [recommend][recommend-helm] using Helm to install an ACK service controller.
 
-The GitHub Issue **WILL BE** associated with a GitHub Milestone indicating the
-target date for the `DEVELOPER PREVIEW` release of that service's controller.
+[recommend-helm]: https://aws-controllers-k8s.github.io/community/user-docs/install/#helm-recommended
 
-## BUILD
+If you want to only install ACK service controllers that have been determined
+to be stable by the ACK contributor team and the associated AWS service team,
+you may choose to install the Helm chart for an ACK service controller that has
+been tagged with the `-stable` suffix -- e.g. `elasticache-stable`.
 
-The `BUILD` status indicates that the ACK service controller for the AWS
-service is **actively being built** in preparation for `DEVELOPER PREVIEW`
-release of that ACK service controller.
+This `-stable` artifact tag points to a Helm chart that has configuration values that
+have been tested with a specific SemVer Docker image for the ACK service
+controller. Typically these tests are "soak" tests and allow the team
+maintaining that ACK controller's source code to have a high degree of
+confidence in the controller's long-running operation.
 
-It is in the `BUILD` status that we will identify those **AWS service API
-resources** that will be supported in `DEVELOPER PREVIEW` and which will be
-supported by the `GENERALLY AVAILABLE` release of the controller.
+!!! "important"
+    Please note that not all ACK service controllers have Helm chart that have
+    been tagged with the `-stable` suffix.
 
-!!! note "what do we mean by 'AWS service API resources'?
-    An *AWS service API resource* is a top-level object that can be created by
-    a particular AWS service API. For example, an SNS Topic or an S3 Bucket.
-    Some service APIs have multiple top-level resources; SNS, for instance, has
-    Topic, PlatformApplication and PlatformEndpoint top-level resources that
-    may be created.
+## Maintenance Phases
 
-## DEVELOPER PREVIEW
+As noted above, individual ACK service controllers all use Semantic Versioning
+("X.Y.Z") in order to signal breaking interface changes. However, each
```

Comment:
How would the version be incremented?
Lets take following scenario:
```
		"main branch"
		|
		commit1
		|	\
		|	"v0.0.1" created (preview)
		commit2
		|	\
		|	v0.0.2" created (preview)
		|
		commit3 -> "stable branch" created
		|			\
		|			"v0.0.3-stable" (GA) created
		commit4
		|	\
		|	"v0.0.4" created (preview)
		|
		commit5
		(breaking change)
		|	\
		|	"v1.0.0" created (preview)
		|
		commit6 -> "stable branch" rebased
		|			\
		|			"v1.0.1-stable" (GA) created
```
In this scenario, what if `v0.0.3-stable` has a critical bug which needs to be patched. Since the next `-stable` version introduced the breaking change and has the major version 1.x.x; should the patch to `v0.0.3-stable` level of source code be released as: `v0.0.5-stable` to avoid forcing consumers to upgrade to a breaking change based release to receive the patch?

Reply:
See above. Docker image tags are never `-stable`. Only a single Helm Chart will ever be tagged with the `-stable` tag, and it is a Helm Chart that references a tested SemVer Docker image for the controller.

Reply:
In the above scenario consider for example "v0.0.4" represents the SemVer; so image is named as "elasticache-v0.0.4"; the helm chart is tagged as "elasticache-v0.0.4". And that "v0.0.3-stable" represents SemVer v0.0.3 where the image is named as "elasticache-v0.0.3" and the helm chart is tagged as "elasticache-v0.0.3-stable".

Please respond considering the above.

Reply:
@kumargauravsharma not entirely sure I'm following your question, so let me outline how I envision the Git branching and Git tagging to happen over time for a service controller repo.

1) The `HEAD` of the `main` Git branch will always point to the latest commit to the latest major version series of the controller.

2) As releases are made, a Git tag with SemVer format will be placed on the Git repository that point to a specific commit SHA1.

If the latest major version series is "0" and before we've determined that anything is "stable" yet, we might see a commit history that looks like this:

(note, I'm using single-letter sequential commit IDs instead of SHA1s for brevity...)

```
commit A
|
commit B <- git tag `v0.0.1`
|
commit C
|
commit D <- git tag `v0.0.2`
|
commit E
| 
commit F  <- git tag `v0.0.3`
|
commit G
|
commit H <-- main/HEAD
```

3) The maintainer team runs a series of soak tests on the code contained in commit F (tagged `v0.0.3`) and are confident this code is good for production use. The maintainer team then creates a new branch called `stable`, based from commit F:

```
git checkout -b stable v0.0.3
```

which will result in the following:

```
commit A
|
commit B <- git tag `v0.0.1`
|
commit C
|
commit D <- git tag `v0.0.2`
|
commit E
| 
commit F  <- git tag `v0.0.3`
|       \
|        <- stable/HEAD
|
commit G
|
commit H <-- main/HEAD
```

The maintainer team runs `scripts/build-controller-release.sh` and `scripts/helm-publish-chart.sh` which ends up create a Helm Chart with a Deployment that has an image reference of `public.ecr.aws/aws-controllers-k8s/controller:elasticache-v0.0.3`. The Helm Chart artifact tag will be `public.ecr.aws/aws-controllers-k8s/chart:elasticache-v0-stable`.

4) More commits are added to the `main` branch, and new release tags are added:

```
commit A
|
commit B <- git tag `v0.0.1`
|
commit C
|
commit D <- git tag `v0.0.2`
|
commit E
| 
commit F  <- git tag `v0.0.3`
|       \
|        <- stable/HEAD
|
commit G
|
commit H <-- git tag `v0.1.1`
|
commit J <-- git tag `v0.1.2`
|
commit K
|
commit L
|
commit M <-- main/HEAD
```

5) The maintainer team runs soak tests on the `v0.1.2` code and finds it is production ready. The then `git rebase` the stable branch to point to the `v0.1.2` commit:

```
git checkout stable
git rebase -i v0.1.2
```

the commit history will then look like this:

```
commit A
|
commit B <- git tag `v0.0.1`
|
commit C
|
commit D <- git tag `v0.0.2`
|
commit E
| 
commit F  <- git tag `v0.0.3`
|
commit G
|
commit H <-- git tag `v0.1.1`
|
commit J <-- git tag `v0.1.2`
|       \
|        <- stable/HEAD
|
commit K
|
commit L
|
commit M <-- main/HEAD
```

When they rebuild their Helm Charts and publish, the Helm Chart with the `elasticache-v0-stable` tag will be replaced with the Helm Chart that now has its Deployment using the `elasticache-v0.1.2` controller image.

6) Now the maintainer team decides to make changes to the controller that break public interfaces, and thus require a major version bump. The commit history now looks like this:

```
commit A
|
commit B <- git tag `v0.0.1`
|
commit C
|
commit D <- git tag `v0.0.2`
|
commit E
| 
commit F  <- git tag `v0.0.3`
|
commit G
|
commit H <-- git tag `v0.1.1`
|
commit J <-- git tag `v0.1.2`
|       \
|        <- stable/HEAD
|
commit K
|
commit L
|
commit M <-- git tag `v1.0.0`
|
commit N <-- git tag `v1.0.1`
|
commit O <- main/HEAD
```

7) After some time, the maintainer team decides that the `v1.0.1` code is good for production use and they want to create a stable Helm Chart pointing to this controller image. Because the existing `stable` Git branch will be pointing to "0"-series controller code, the maintainer team will need to create a `stable-v0` Git branch from the existing `stable` branch:

```
git checkout -b stable-v0 stable && git push upstream stable-v0
```

and then they are free to modify the `stable` branch to point to the latest major version series' stable commit:

```
git checkout stable
git rebase -i v1.0.1
```

Our git history would then look like this:

```
commit A
|
commit B <- git tag `v0.0.1`
|
commit C
|
commit D <- git tag `v0.0.2`
|
commit E
| 
commit F  <- git tag `v0.0.3`
|
commit G
|
commit H <-- git tag `v0.1.1`
|
commit J <-- git tag `v0.1.2`
|       \
|        <- stable-v0/HEAD
|
commit K
|
commit L
|
commit M <-- git tag `v1.0.0`
|
commit N <-- git tag `v1.0.1`
|       \
|        <- stable/HEAD
|
commit O <- main/HEAD
```

and rebuild the Helm Chart. The new Helm Chart will have the artifact tag of `chart:elasticache-v1-stable` and will point to the `controller:elasticache-v1.0.1` image.

Reply:
Thanks @jaypipes 

Mapping my query to this scenario: After step 7 is done; if maintainer team needs to fix a critical vulnerability discovered in the controller binary (i.e. elasticache-v0.1.2 controller image) pointed by`elasticache-v0-stable` Helm chart then because the fix for that vulnerability can not be on "main" branch (as the main/HEAD has breaking changes compared to v0).
In this scenario, the service team would need to have the 'fix' checked in into 'stable-v0' code branch and git tag corresponding commit id (from stable-v0 branch) as `v0.1.3` and generate release artifacts from this commit level.

In other words, the fix will not be checked in into "main" for v0 series. Though it can later be merged from 'stable-v0`) branch to "main" branch to have it available in v1 series as well.
Please validate if the above understanding is correct.

Reply:
> Thanks @jaypipes
> 
> Mapping my query to this scenario: After step 7 is done; if maintainer team needs to fix a critical vulnerability discovered in the controller binary (i.e. elasticache-v0.1.2 controller image) pointed by`elasticache-v0-stable` Helm chart then because the fix for that vulnerability can not be on "main" branch (as the main/HEAD has breaking changes compared to v0).
> In this scenario, the service team would need to have the 'fix' checked in into 'stable-v0' code branch and git tag corresponding commit id (from stable-v0 branch) as `v0.1.3` and generate release artifacts from this commit level.

Yes, above is 100% correct, @kumargauravsharma 

> In other words, the fix will not be checked in into "main" for v0 series. Though it can later be merged from 'stable-v0`) branch to "main" branch to have it available in v1 series as well.

Well, not merged, but cherry-picked from the stable-v0 branch into main, yes.

Best,
-jay